src/brain.coffee

```diff
@@ -148,6 +148,21 @@ class Brain extends EventEmitter
 
     matchedUsers
 
+  # Public: Get all users that contain the provided query string
+  # in any of their properties.
+  #
+  # Returns an Array of User instances matching the search criteria.
+  usersForAny: (query) ->
+    query = query.toLowerCase()
+    matchedUsers = []
+    for key, user of (@data.users or {})
+      for property, value of user
+        if typeof value == 'string'
+          if value.toLowerCase().indexOf(query) isnt -1
+            matchedUsers.push user
+            break
+    return matchedUsers
```

Comment:
I personally believe explicit returns are better than implicit ones except for single-line lambdas for example. Even for a terse ruby-influenced lang like coffeescript.

But given that the project uses implicit ones everywhere it makes sense to stick to the convention. I've fixed it.